modules/core/src/main/scala/gem/parser/Misc.scala

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.parser
+
+import atto._, Atto._
+import scala.annotation.tailrec
+
+/** General-purpose parsers and combinators that aren't provided by atto. */
+trait MiscParsers {
+
+  /** Parser for a hyphen. */
+  val hyphen: Parser[Unit] =
+    char('-').void
+
+  /** Parser for one or more spaces. */
+  val spaces1: Parser[Unit] =
+    skipMany1(char(' '))
+
+  /** Parser for a non-whitespace string. */
+  val nonWhitespace: Parser[String] =
+    stringOf(elem(c => !c.isWhitespace))
```

Comment:
Not that it matters to us performance wise, but I'm curious to know if there is an easy way to check equality for whole string buffers instead of going char by char.

That's one of the main advantages of the Haskell's `attoparsec` with respect to the original `Parsec`.